contracts/access/Bouncer.sol

```diff
@@ -146,14 +160,27 @@ contract SignatureBouncer is Ownable, RBAC {
    * and then recover the signature and check it against the bouncer role
    * @return bool
    */
-  function isValidDataHash(bytes32 hash, bytes _sig)
+  function isValidDataHash(bytes32 _hash, bytes _sig)
     internal
     view
     returns (bool)
   {
-    address signer = hash
+    // if the sender is a delegate AND supports isValidSignature, we delegate validation to them
+    // this allows someone who wants custom validation logic (perhaps they check another contract's state)
+    // to code their own delegate. The users then submit actions to the delegate contract, which proxies
+    // them to this contract, which then pings-back to check if it's cool.
+    // This is also useful for contracts-as-identities: your identity contract implements isValidSignature
+    // and can then recover the signer and check it against the whitelisted ACTION keys.
+    if (hasRole(msg.sender, ROLE_DELEGATE) &&
+        msg.sender.supportsInterface(InterfaceId_BouncerDelegate)) {
```

Comment:
Why use `InterfaceId_BouncerDelegate`? Isn't 165 used to make sure a given contract has implemented the proper function when it is called? I believe the right call would be `supportsInterface(InterfaceId_IsValidSignature)` to make sure the bouncer contract supports `isValidSignature()`.

Reply:
In this case we're asking the sender (which is the supposed delegate) if they support the correct function. In this case we're calling `isValidSignature` on the Delegate, not the Bouncer, so I think this is correct.

Reply:
Sorry, I when I meant Bouncer contract, I meant the contract that has a bouncer role, i.e. the delegate. Why would the delegate contract support the `InterfaceId_BouncerDelegate`? Does it need to? All the delegate contract needs to have is `isValidSignature()`, no?

Reply:
That's what the `InterfaceId_BouncerDelegate` is testing, yeah :) — it checks the delegate for the isValidSignature method implementation.

Reply:
Ah, yes. We are on the same page then! Just saw the comment in `IBouncerDelegate.sol`.